builtin/logical/pki/acme/state.go

```diff
@@ -0,0 +1,152 @@
+package acme
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/framework"
+)
+
+// How long nonces are considered valid.
+const nonceExpiry = 15 * time.Minute
+
+type ACMEState struct {
+	nextExpiry atomic.Int64
+	nonces     *sync.Map // map[string]time.Time
+}
```

Comment:
This should probably be addressed, nextExpiry should be a pointer type.

Reply:
I don't understand why nextExpiry should have a pointer type? For one, we don't use `uber-go/atomics`, for two, [the type definition](https://pkg.go.dev/sync/atomic#Int64) doesn't give any indication of this, unlike [`sync.Map` which does](https://pkg.go.dev/sync#Map), and for three, the [direct functions](https://pkg.go.dev/sync/atomic#AddUint64) don't give any indication of this either.

In the case of a locking data type, which this is not, it is clear that copying a lock leaves it in a weird state (as now either copy can be locked/unlocked independently of the other and a pointer should definitely be used for clarity).

But for an atomic data type, I'm not sure as to why this truly holds, _given_ we're not using this as a lock, but instead using this as a thread-safe way of reading/writing a time value? That time value is still applicable across copies, and one would want to write in to each separately. My 2c. :-)

Reply:
Ah interesting, the type definition does have a `noCopy` declaration: https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/sync/atomic/type.go;l=90-95

But it otherwise calls out to the non-copy-prevented helpers I mentioned earlier, which is weird...

Reply:
Yeah looking around the PKI code base, we are already inconsistent it seems :(

Reply:
Fixed, thanks!